lake/pool.go

```diff
@@ -202,6 +200,20 @@ func (p *Pool) LookupTags(ctx context.Context, tags []ksuid.KSUID) ([]ksuid.KSUI
 	return ids, nil
 }
 
+func (p *Pool) SnapshotOf(ctx context.Context, tag ksuid.KSUID) (*commit.Snapshot, error) {
+	if tag == ksuid.Nil {
+		return p.log.Head(ctx)
+	}
+	snap, ok, err := p.log.SnapshotOfCommit(ctx, 0, tag)
+	if err != nil {
+		return nil, fmt.Errorf("tag does not exist: %s", tag)
+	}
+	if !ok {
+		return nil, fmt.Errorf("commit tag was previously deleted: %s", tag)
```

Comment:
```suggestion
		return nil, fmt.Errorf("tag was previously deleted: %s", tag)
```

Reply:
The tag is a commit tag not an object tag so I think the clarification is helpful.